classes/class.ilMumieTaskGradeOverrideService.php

```diff
@@ -44,8 +44,8 @@ private static function loadOverriddenGrade($user_id, $task)
         "usr_id = " . $ilDB->quote($user_id, "text") .
         " AND " .
         "task_id = " . $ilDB->quote($task->getId(), "integer");
-        $result = $ilDB->query($query);
-        return $ilDB->fetchAssoc($result)[self::NEW_GRADE];
+        $assoc = $ilDB->fetchAssoc($ilDB->query($query));
+        return is_null($assoc) ? $assoc : $assoc[self::NEW_GRADE];
```

Comment:
If I write an functional equivalent of your suggestion, it will look like this:

```
        if (!$assoc) {
            //Offsetting on a value of type null wont work
            return $assoc[self::NEW_GRADE];
        } else {
            //We don't return the specific value but the entire array
            return $assoc;
        }
```

But we want it to behave like this:

```
        if ($assoc) {
            //Return a specific value of the array, if the array is defined
            return $assoc[self::NEW_GRADE];
        } else {
            //Return null, if the array is null
            return $assoc;
        }
```

I don't think we can write a good version of this code with a null coalescing opperator

Reply:
you are right :)